ood-portal-generator/lib/ood_portal_generator/dex.rb

```diff
@@ -104,7 +104,8 @@ def tls_key
     end
 
     def issuer
-      "#{protocol}#{servername}:#{port}"
```

Comment:
Ya know I think this change may do the trick. Just give precedence to the existing `proxy_server` if it exists.

change 
https://github.com/OSC/ondemand/blob/17d924b82068a83c5efd8d00a1f413032f0752a0/ood-portal-generator/lib/ood_portal_generator/dex.rb#L82-L84

to 

```ruby
def servername
  @view.proxy_server || @view.servername || OodPortalGenerator.fqdn
end
```

with a test case or 2 to be sure that when you set proxy_server it's everywhere and if set servername _it's_ everywhere.